src/lesson5-homework.ts

```diff
@@ -1,230 +1,488 @@
+
+        constructor(name: string, width: number, height: number, color: string, space: number,) {
+            this.color = color;
+            this.height = height;
+            this.width = width;
+            this.name = name
+            this.space = space
+
+        }
+
+        getStringifiedParams(): string {
+            return `Class: ${this.constructor.name},
+             name: "${this.name},
+             width: ${this.width},
+             height: ${this.height},
+             color: ${this.color},
+             space: ${this.space}`
+        }
+
+        install(): void {
+            console.log('widget is installed')
+        }
+
+        resize(): void {
+            this.width++
+            this.height++
+        }
+    }
+
+    const mobWid = new MobileWidget('Modbile widget', 600, 800, 'black', 5)
+    console.log('Exercise 28', mobWid.getStringifiedParams())
+
+    // TODO: declare class DesktopAndMobileWidget which implements IDesktopWidget and IMobileWidget
+    class DesktopAndMobileWidget implements IDesktopWidget, IMobileWidget, IWidgetPrintable {
+        color: string;
+        height: number;
+        name: string;
+        space: number;
+        width: number;
+        os: string;
+
+        constructor(name: string, width: number, height: number, color: string, space: number, os: string) {
+            this.color = color;
+            this.height = height;
+            this.width = width;
+            this.space = space;
+            this.os = os;
+            this.name = name;
+        }
+
+        getStringifiedParams(): string {
+            return `Class: ${this.constructor.name},
+             name: "${this.name},
+             width: ${this.width},
+             height: ${this.height},
+             color: ${this.color},
+             space: ${this.space}
+             os: ${this.os}`
+        }
+
+        install(): void {
+        }
+
+        open(): void {
+        }
+
+        resize(): void {
+        }
+    }
+
+    const deskNMobWid = new DesktopAndMobileWidget('Hybrid', 55, 110, 'blue', 15, 'IOS')
+    console.log('Exercise 28 ', deskNMobWid.getStringifiedParams())
+
+
+    // TODO: declare interface IWidgetPrintable which has toString method
+    interface IWidgetPrintable {
+        // toString method - returns string
+        /**
+         * Returns data in string format
+         * note: I decided change name because toString is default method.
+         * In this case typescript does not push me as a developer implement this function.
+         * Obviously because this function is present in the base prototype
+         * i.e. accessible through any object.
+         * It looks unsafe for me in case developer will not implement function in the child class
+         * typescript has no reaction on it. So I am not sure that I can rely on this function
+         */
+        getStringifiedParams(): string
+
+    }
+
+    // TODO: add IWidgetPrintable to each of the classes above
```

Comment:
good observation - yes typescript is not forcing you to implement toString method
nevertheless - toString is default implementation and there is a good reason why this is needs to be implemented 
main - it is used when object is used "as a string" 
so you have to write console.log('Exercise 28 ', deskNMobWid.getStringifiedParams())
but if toString is implemented - the call would simple be 
console.log('Exercise 28 ', deskNMobWid)